packages/react/src/xcss-prop/index.ts

```diff
@@ -27,19 +25,29 @@ type XCSSPseudos<
     : never;
 };
 
+type XCSSMediaQuery<
+  TAllowedProperties extends keyof CSSProperties,
+  TAllowedPseudos extends CSSPseudos,
+  TAllowedMediaQueries extends string,
+  TRequiredProperties extends { requiredProperties: TAllowedProperties },
+  TSchema
+> = {
+  [Q in `@media ${TAllowedMediaQueries}`]?:
+    | MarkAsRequired<
+        XCSSItem<TAllowedProperties, Q extends keyof TSchema ? TSchema[Q] : object>,
+        TRequiredProperties['requiredProperties']
+      >
+    | XCSSPseudos<TAllowedProperties, TAllowedPseudos, TRequiredProperties, TSchema>;
+};
+
 /**
  * These APIs we don't want to allow to be passed through the `xcss` prop but we also
  * must declare them so the (lack-of a) excess property check doesn't bite us and allow
  * unexpected values through.
  */
 type BlockedRules = {
   selectors?: never;
-} & {
-  /**
-   * We currently block all at rules from xcss prop.
-   * This needs us to decide on what the final API is across Compiled to be able to set.
-   */
-  [Q in CSS.AtRules]?: never;
+  '@media'?: never;
```

Comment:
do we want to block other at rules?

Reply:
Maybe. What's your thoughts?

Reply:
Hi it's me again: Yes we should.

Reply:
yeah e.g. we want to make sure people don't start using `@keyframes` cuz we have a specialised `keyframes` function for that, and we don't want people using `@import` and `@font-face` in unexpected places... so a whitelist system for at-rules would make sense